app/middleware/get-service-and-gateway-account.middleware.js

```diff
@@ -71,47 +69,35 @@ function getService (user, serviceExternalId, gatewayAccount, correlationId) {
     // Since it is only card currently supported, value is always 'true'
     service.hasCardGatewayAccount = true
     return service
-  } else {
-    const logContext = {}
-    logContext[keys.USER_EXTERNAL_ID] = user.externalId
-    logContext[keys.SERVICE_EXTERNAL_ID] = serviceExternalId
-    if (gatewayAccount) {
-      logContext[keys.GATEWAY_ACCOUNT_ID] = gatewayAccount.gateway_account_id
-    }
-    logContext[keys.CORRELATION_ID] = correlationId
-    logger.info('Service not found for user', logContext)
-    throw new NotFoundError('Service not found for user')
   }
 }
 
-module.exports = async function (req, res, next) {
+module.exports = async function getServiceAndGatewayAccount (req, res, next) {
   try {
-    const correlationId = req.correlationId
-    const serviceExternalId = req.params[paths.keys.SERVICE_EXTERNAL_ID]
-    const gatewayAccountExternalId = req.params[paths.keys.GATEWAY_ACCOUNT_EXTERNAL_ID]
-
-    if (!serviceExternalId && !gatewayAccountExternalId) {
-      const logContext = {}
-      logContext[keys.CORRELATION_ID] = correlationId
-      logger.info('Could not resolve gateway account external ID or service external ID from request params')
-      throw new Error('Could not resolve gateway account external ID or service external ID from request params')
-    }
+    if (req.user) {
```

Comment:
could consider reversing this condition to reduce indentation but that's a style choice